src/util/string/hash.js

```diff
@@ -0,0 +1,12 @@
+define(function() {
+
+// Based on http://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript-jquery
+return function( str ) {
+	return [].reduce.call( str, function( hash, i ) {
```

Comment:
Dunno if there's a difference in practice. Function calls have a price used to be a good rule of thumb. What I find more important here: I find the `for` loop easier to understand than `reduce()`.